nautobot/core/views/__init__.py

```diff
@@ -117,13 +42,42 @@ def get(self, request):
         context.update(
             {
                 "search_form": SearchForm(),
-                "stats": stats,
-                "job_results": job_results[:10],
-                "changelog": changelog[:15],
                 "new_release": new_release,
             }
         )
 
+        # Loop over homepage layout to collect all additional data and create custom panels.
+        for column_name, column_details in registry["homepage_layout"]["columns"].items():
+            for panel_name, panel_details in column_details["panels"].items():
+                for item_name, item_details in panel_details["items"].items():
+
+                    if item_details.get("custom_code"):
+                        # Collect all custom data using callback functions.
+                        for key, function in item_details.get("custom_data", {}).items():
+                            context[key] = function(request)
+                        # Create standalone template
+                        sub_template = Template(item_details["custom_code"].strip())
+                        sub_context = RequestContext(request, context)
+                        # Store rendered sub-template in registry.
+                        registry["homepage_layout"]["columns"][column_name]["panels"][panel_name]["items"][item_name][
+                            "rendered_html"
+                        ] = sub_template.render(sub_context)
+
+                    elif item_details.get("model"):
+                        # If there is a model attached collect object count.
+                        registry["homepage_layout"]["columns"][column_name]["panels"][panel_name]["items"][item_name][
+                            "count"
+                        ] = (item_details["model"].objects.restrict(request.user, "view").count())
+
+                    elif item_details.get("items"):
```

Comment:
So is the idea that `custom_code`, `model`, and `items` are mutually-exclusive keys on `item_details`? If so, then IMO that needs to be enforced as part of the data validation when this is registered in the first place.